projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php

```diff
@@ -286,30 +303,63 @@ function get_checklist_task( $task ) {
  *
  * @param string $checklist_slug Checklist slug.
  *
- * @return array Associative array with checklist task
- *               or empty array if checklist slug is not found.
+ * @return Task[] Collection of tasks for a given checklist
  */
-function build_checklist( $checklist_slug ) {
-	$checklist = array();
-	if ( null === ( get_checklist_definitions()[ $checklist_slug ] ) ) {
-		return $checklist;
-	}
-	foreach ( get_checklist_definitions()[ $checklist_slug ] as $task_id ) {
-		$checklist[] = get_task_definitions()[ $task_id ];
+function get_launchpad_checklist_by_checklist_slug( $checklist_slug ) {
+	if ( ! $checklist_slug ) {
+		return array();
 	}
-	return $checklist;
+
+	$launchpad_task_lists = Launchpad_Task_Lists::get_instance();
+	return $launchpad_task_lists->build( $checklist_slug );
 }
 
+// TODO: Write code p2 post or dotcom post
 /**
- * Returns launchpad checklist by checklist slug.
+ * Wrapper that registers a launchpad checklist.
  *
- * @param string $checklist_slug Checklist slug.
+ * @param Task_List $task_list Task list definition.
  *
- * @return array Associative array with checklist task data
+ * @return bool True if successful, false otherwise.
  */
-function get_launchpad_checklist_by_checklist_slug( $checklist_slug ) {
-	if ( ! $checklist_slug ) {
-		return array();
+function register_launchpad_task_list( $task_list ) {
+	$launchpad_task_lists = Launchpad_Task_Lists::get_instance();
+	return $launchpad_task_lists->register_task_list( $task_list );
+}
+
+/**
+ * Wrapper that registers a launchpad checklist.
+ *
+ * @param Task $tasks Collection of Task definitions.
+ *
+ * @return bool True if successful, false otherwise.
+ */
+function register_launchpad_tasks( $tasks ) {
+	$launchpad_task_lists = Launchpad_Task_Lists::get_instance();
+	return $launchpad_task_lists->register_tasks( $tasks );
+}
+
+/**
+ * Wrapper that registers a launchpad checklist.
+ *
+ * @param Task $task Task definition.
+ *
+ * @return bool True if successful, false otherwise.
+ */
+function register_launchpad_task( $task ) {
+	$launchpad_task_lists = Launchpad_Task_Lists::get_instance();
+	return $launchpad_task_lists->register_task( $task );
+}
+
+/**
+ * Registers all default launchpad checklists
+ */
+function register_default_checklists() {
+	foreach ( get_checklist_definitions() as $checklist ) {
+		register_launchpad_task_list( $checklist );
 	}
-	return build_checklist( $checklist_slug );
+
+	register_launchpad_tasks( get_task_definitions() );
```

Comment:
I would like to circle back and refactor this to explicitly call `register_launchpad_checklist` for each individual task and task list but I don't mind starting out in this state.

Reply:
Sounds good 👍

Reply:
It might also be good to [prefix function names](https://developer.wordpress.org/plugins/plugin-basics/best-practices/#prefix-everything) to avoid potential collisions on WoA sites